Reflection/NPC.simba

```diff
@@ -201,3 +222,234 @@ begin
+  Definition: RSNPCDefinition;
+  Tiles: Array of RSTile;
+  Tile: RSTile;
+  MSBounds: TBox;
+  I: Integer;
+begin
+  NPCs := RSClient.AllNPCs;
+  MSBounds := R_MainscreenBounds;
+  for I := 0 to High(NPCs) do
+  begin
+    Definition := NPCs[I].Definition;
+    Tile := NPCs[I].LocalTile.TileToMS;
+    if ((Definition.Name = Name)  and (Tile.X > MSBounds.X1) and (Tile.X < MSBounds.X2) and (Tile.Y > MSBounds.Y1) and (Tile.Y < MSBounds.Y2)) then
+      Tiles += NPCs[I].Tile;
+    Definition.Free;
+  end;
+  Result := Tiles;
+  RSTypeArray(NPCs).Free;
+end;
+
+Function R_NPCGetTileMS(ID: Int32): Array of RSTile;  overload;
+var
+  NPCs: Array of RSNPC;
+  Definition: RSNPCDefinition;
+  Tiles: Array of RSTile;
+  Tile: RSTile;
+  MSBounds: TBox;
+  I: Integer;
+begin
+  NPCs := RSClient.AllNPCs;
+  MSBounds := R_MainscreenBounds;
+  for I := 0 to High(NPCs) do
+  begin
+    Definition := NPCs[I].Definition;
+    Tile := NPCs[I].LocalTile.TileToMS;
+    if ((Definition.ID = ID)  and (Tile.X > MSBounds.X1) and (Tile.X < MSBounds.X2) and (Tile.Y > MSBounds.Y1) and (Tile.Y < MSBounds.Y2)) then
+      Tiles += NPCs[I].Tile;
+    Definition.Free;
+  end;
+  Result := Tiles;
+  RSTypeArray(NPCs).Free;
+end;
+
+(*
+R_NPCGetMSCoords
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Returns the Pixel Coordinates of the given NPC if they are on screen.
+
+Example:
+
+   writeln(R_NPCGetMSCoords('Chicken'));
+*)
+Function R_NPCGetMSCoords(Name: String): TPointArray;
+var
+  NPCs: Array of RSNPC;
+  Definition: RSNPCDefinition;
+  Tiles: TPointArray;
+  Tile: RSTile;
+  MSBounds: TBox;
+  I: Integer;
+begin
+  NPCs := RSClient.AllNPCs;
+  MSBounds := R_MainscreenBounds;
+  for I := 0 to High(NPCs) do
+  begin
+    Definition := NPCs[I].Definition;
+    Tile := NPCs[I].LocalTile.TileToMS;
+    if ((Definition.Name = Name)  and (Tile.X > MSBounds.X1) and (Tile.X < MSBounds.X2) and (Tile.Y > MSBounds.Y1) and (Tile.Y < MSBounds.Y2)) then
+      Tiles.Append(Tile);
+    Definition.Free;
+  end;
+  Result := Tiles;
+  RSTypeArray(NPCs).Free;
+end;
+
+Function R_NPCGetMSCoords(ID: Int32): TPointArray; overload;
+var
+  NPCs: Array of RSNPC;
+  Definition: RSNPCDefinition;
+  Tiles: TPointArray;
+  Tile: RSTile;
+  MSBounds: TBox;
+  I: Integer;
+begin
+  NPCs := RSClient.AllNPCs;
+  MSBounds := R_MainscreenBounds;
+  for I := 0 to High(NPCs) do
+  begin
+    Definition := NPCs[I].Definition;
+    Tile := NPCs[I].LocalTile.TileToMS;
+    if ((Definition.ID = ID)  and (Tile.X > MSBounds.X1) and (Tile.X < MSBounds.X2) and (Tile.Y > MSBounds.Y1) and (Tile.Y < MSBounds.Y2)) then
+      Tiles.Append(Tile);
+    Definition.Free;
+  end;
+  Result := Tiles;
+  RSTypeArray(NPCs).Free;
+end;
+
+(*
+R_NPCGetMSCoordsSort
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Returns the Pixel Coordinates of the given NPC if they are on screen
+  as a sorted array from the middle of the screen.
+
+Example:
+
+   writeln(R_NPCGetMSCoordsSort('Chicken'));
+*)
+Function R_NPCGetMSCoordsSort(Name: String): TPointArray;
```

Comment:
I'm thinking here you could do something like:

```
Function R_NPCGetMSCoordsSort(Name: String): TPointArray;
var
  Tiles : TPointArray;
begin
  Tiles := R_NPCGetMSCoords(Name);
  Tiles.sort(R_MainscreenBounds.Middle);
  Result := Tiles;
end;
```

Reply:
Haha. I implemented that but must've accidentally written over my changes.

Reply:
I had that in there but must've accidentally written over my previous save.